backend/core/src/main/java/org/sonarsource/sonarlint/core/reporting/FindingReportingService.java

```diff
@@ -153,6 +156,7 @@ public void reportTrackedFindings(String configurationScopeId, UUID analysisId,
     var isMQRMode = severityModeService.isMQRModeForConnection(connectionId);
     var aiCodeFixFeature = effectiveBinding.flatMap(aiCodeFixService::getFeature);
     var issuesToRaise = getIssuesToRaise(issuesToReport, newCodeDefinition, isMQRMode, aiCodeFixFeature);
+    aiCodeFixFeature.ifPresent(f -> previouslyRaisedFindingsRepository.countAiFixableIssuesForTelemetry(configurationScopeId, issuesToRaise, telemetryService));
```

Comment:
Same idea here, the responsibility of the repository is not to do things related to telemetry, it's not its business